base/server/cmscore/src/com/netscape/cmscore/ldapconn/LdapAnonConnFactory.java

```diff
@@ -295,6 +320,17 @@ public synchronized LDAPConnection getConn(boolean waitForConn)
         }
         //This is the end of the fix for Bugzilla #630176
 
+        try {
+            // Before returning the connection, set the SIZELIMIT option; this
+            // ensures that if the connection is recycled and the previous owner
+            // changed the SIZELIMIT option to a different value, the next owner
+            // always starts with the default.
+            conn.setOption(LDAPv2.SIZELIMIT, mMaxResults);
+        } catch (LDAPException e) {
+            // Ignore the exception; it isn't important if we can't set our
+            // SIZELIMIT value.
```

Comment:
We should avoid ignoring exceptions like this, especially without logging any warnings. In general, if the admin provides a configuration parameter, but somehow the parameter cannot be applied properly, we do want the admin to investigate the problem, otherwise the problem may appear somewhere else. To provide more flexibility, I think we should allow value 0 for no limit, then positive value for the actual limit. So if the limit cannot be set this method should throw an exception, but the admin will have an option to bypass it if necessary by setting it to 0.

Reply:
There is no way for `conn.setOption` to *not* set a maximum results option parameter at this point. It doesn't even check for negative values. 

[`conn.setOption(int option, Object value)`](https://github.com/cipherboy/ldap-sdk/blob/master/java-sdk/ldapjdk/netscape/ldap/LDAPConnection.java#L4441) calls [`conn.setOption(option, value, m_defaultConstraints)`](https://github.com/cipherboy/ldap-sdk/blob/master/java-sdk/ldapjdk/netscape/ldap/LDAPConnection.java#L4449) as `LDAPv2.SIZELIMIT != LDAPv2.PROTOCOL_VERSION`.

This three-argument version is a big switch statement wrapped in a `try/catch (ClassCastException)`. The catch just reraises the `ClassCastException` as a `LDAPException`. The default branch of the switch also throws a `LDAPException`, stating that an unknown option was set. We know we're in the `LDAPv2.SIZELIMIT` case statement, which calls `cons.setMaxResults(...)` (where `cons == this.m_defaultConstraints`) and then returns. [`cons.setMaxResults(int maxResults)`](https://github.com/cipherboy/ldap-sdk/blob/master/java-sdk/ldapjdk/netscape/ldap/LDAPSearchConstraints.java#L296-L303) does not throw anything. 

Therefore, there is no way for this to throw an LDAPException: `LDAPv2.SIZELIMIT` is used in both places, so `LDAPv2.SIZELIMIT == LDAPv2.SIZELIMIT` ensuring we don't hit the default branch. If we fail to cast `int -> Integer` (since we're using a primitive `int`, which can't be `null` and can't be a generic `Object` either)... the JVM is having problems and we have much bigger issues than trying to log this failure.